### DISCUSSION TITLE
[Proposal:] Modify static abstract interfaces for functions.

### DISCUSSION THREAD

**Discussion by RVShershnev** [dotnet/runtime#90737] [General] [answered]
I try create math library for fuzzy numbers and I face with problems.

For example, I need:
```csharp
public readonly partial struct Triangular  :  IPowerFunctions<Triangular, double, Triangular>
{
    public static Triangular Pow(Triangular number, double power)
} 
```

I proposal the following suggestion in solving the problem.
1) Add interface for IAbsFunctions<TSelf, TResult> as `Math.Abs(...)`.
2) Change interface [IPowerFunctions<TSelf>](https://learn.microsoft.com/ru-ru/dotnet/api/system.numerics.ipowerfunctions-1) to `IPowerFunctions<TSelf, TPower, TResult>`. 
3) Change interface [IExponentialFunctions<TSelf>](https://learn.microsoft.com/ru-ru/dotnet/api/system.numerics.iexponentialfunctions-1) to `IExponentialFunctions<TSelf, TResult>` 
4) Change interface [IRootFunctions<TSelf>](https://learn.microsoft.com/ru-ru/dotnet/api/system.numerics.irootfunctions-1) to `IRootFunctions<TSelf, TResult>`
5) Change interface [ILogarithmicFunctions<TSelf>](https://learn.microsoft.com/ru-ru/dotnet/api/system.numerics.ilogarithmicfunctions-1) to `ILogarithmicFunctions<TSelf, TBase, TResult>`
6) Change interface [ITrigonometricFunctions<TSelf>](https://learn.microsoft.com/ru-ru/dotnet/api/system.numerics.itrigonometricfunctions-1) to `ITrigonometricFunctions<TSelf, TResult>`

**About:**
Fuzzy numbers are a type of mathematical concept that allows for values to be expressed in a range rather than a single point.
They are used to represent uncertain or imprecise data, where there is some degree of ambiguity or variability in the measurement.
Fuzzy numbers are typically defined by a membership function that assigns a degree of membership to each value within the range.
This allows for more flexible and nuanced analysis of data, particularly in situations where traditional numerical methods may not be appropriate or effective.

**Comment by HaloFour** [comment, marked as answer]
API changes can be suggested on the [runtime](https://github.com/dotnet/runtime) repository.
